### PR TITLE
Add interface files to modules in library `Pickle types`

### PR DESCRIPTION
**Pull request by rbonichon**
This PR adds minimal interfaces to *all* the modules of the library. 

This allows to clearly see the exported API a little bit more clearly in order to start documenting it.
A side effect is to allow to remove some dead code.

Behavior of the codebase should be unchanged.

### CODE REVIEW

File: src/lib/pickles_types/plonk_verification_key_evals.mli [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+       -> 'comm
+       -> 'comm
+       -> 'comm
+       -> 'comm
+       -> 'comm
+       -> unit )
+     H_list.t
+  -> 'comm t
+
+(** [typ] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/plonk_verification_key_evals.mli:62]: .

File: src/lib/pickles_types/plonk_verification_key_evals.mli [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+val typ :
+     ('a, 'b, 'c) Snarky_backendless.Typ.t
+  -> ('a t, 'b t, 'c) Snarky_backendless.Typ.t
+
+(** [map t ~f] applies [f] to all elements of type ['a] within record [t] and
+    returns the result. In particular, [f] is applied to the elements of
+    {!sigma_comm} and {!coefficients_comm}. *)
+val map : 'a t -> f:('a -> 'b) -> 'b t
+
+(** [map2] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/plonk_verification_key_evals.mli:72]: .

File: src/lib/pickles_types/shifted_value.mli [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  end
+
+  val of_field :
+    (module Field_intf with type t = 'f) -> shift:'f Shift.t -> 'f -> 'f t
+
+  val to_field :
+    (module Field_intf with type t = 'f) -> shift:'f Shift.t -> 'f t -> 'f
+end
+
+(** [Type2] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/shifted_value.mli:92]: .

File: src/lib/pickles_types/vector.mli [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+module Vector_6 : VECTOR with type 'a t = ('a, Nat.N6.n) vec
+
+module Vector_5 : VECTOR with type 'a t = ('a, Nat.N5.n) vec
+
+module Vector_4 : VECTOR with type 'a t = ('a, Nat.N4.n) vec
+
+module Vector_2 : VECTOR with type 'a t = ('a, Nat.N2.n) vec
+
+(** [With_length] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:94]: .

File: src/lib/pickles_types/vector.mli [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+module With_length (N : Nat_intf) : S with type 'a t = ('a, N.n) vec
+
+(** [reduce_exn] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:97]: .

File: src/lib/pickles_types/vector.mli [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+val reduce_exn : ('a, 'n) t -> f:('a -> 'a -> 'a) -> 'a
+
+(** [typ] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:100]: .

File: src/lib/pickles_types/vector.mli [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+val to_list : ('a, 'n) t -> 'a list
+
+(** [zip v1 v2] combines together vectors [v1] and [v2] of length ['b].  *)
+val zip : ('a, 'b) t -> ('c, 'b) t -> ('a * 'c, 'b) t
+
+(** [init n ~f] *)
+val init : 'a Nat.t -> f:(int -> 'b) -> ('b, 'a) t
+
+(** [map] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:122]: .

File: src/lib/pickles_types/vector.mli [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+val map : ('a, 'n) t -> f:('a -> 'b) -> ('b, 'n) t
+
+val map2 : ('a, 'n) t -> ('b, 'n) t -> f:('a -> 'b -> 'c) -> ('c, 'n) t
+
+(** [to_array] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:127]: .

File: src/lib/pickles_types/vector.mli [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+val to_array : ('a, 'b) t -> 'a array
+
+(** [sexp_of_t] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:130]: .

File: src/lib/pickles_types/vector.mli [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+val sexp_of_t :
+     ('a -> Ppx_sexp_conv_lib.Sexp.t)
+  -> 'b
+  -> ('a, 'c) t
+  -> Ppx_sexp_conv_lib.Sexp.t
+
+(** [foldi] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:137]: .

File: src/lib/pickles_types/vector.mli [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+val foldi : ('a, 'b) t -> f:(int -> 'c -> 'a -> 'c) -> init:'c -> 'c
+
+(** [fold] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:140]: .

File: src/lib/pickles_types/vector.mli [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+val fold : ('a, 'n) t -> f:('acc -> 'a -> 'acc) -> init:'acc -> 'acc
+
+val iter : ('a, 'n) t -> f:('a -> unit) -> unit
+
+val iter2 : ('a, 'n) t -> ('b, 'n) t -> f:('a -> 'b -> unit) -> unit
+
+(** [for_all v ~f] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:147]: .

File: src/lib/pickles_types/vector.mli [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+val for_all : ('a, 'n) t -> f:('a -> bool) -> bool
+
+(** [split v ] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:150]: .

File: src/lib/pickles_types/vector.mli [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+val split : ('a, 'n_m) t -> ('n, 'm, 'n_m) Nat.Adds.t -> ('a, 'n) t * ('a, 'm) t
+
+(** [rev v] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:153]: .

File: src/lib/pickles_types/vector.mli [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+val rev : ('a, 'n) t -> ('a, 'n) t
+
+val length : ('a, 'n) t -> 'n Nat.t
+
+val append :
+  ('a, 'n) vec -> ('a, 'm) vec -> ('n, 'm, 'n_m) Nat.Adds.t -> ('a, 'n_m) vec
+
+(** [unsingleton] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:161]: .

File: src/lib/pickles_types/vector.mli [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+val unsingleton : ('a, Nat.z Nat.s) t -> 'a
+
+val trim : 'a 'n 'm. ('a, 'm) vec -> ('n, 'm) Nat.Lte.t -> ('a, 'n) vec
+
+(** [of_array_and_length_exn] *)
```
Comment (mrmr1993) [inline src/lib/pickles_types/vector.mli:166]: .

**Review by mrmr1993 — APPROVED** [review]
Looks good, but can you remove the non-documenting doc-comments?

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/pickles_types/at_most.mli [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+  end
+
+  type 'a t = 'a Stable.V1.t
+
+  include S with type 'a t := 'a t
+end
+
+(** {2 Modules}*)
+
+(** [At_most_2] *)
```
[the remark this answers is not in the document]
Comment (rbonichon) [reply src/lib/pickles_types/at_most.mli]: Done

File: src/lib/pickles_types/at_most.mli [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+module At_most_2 : VERSIONED with type 'a ty = ('a, Nat.N2.n) at_most
+
+module At_most_8 : VERSIONED with type 'a ty = ('a, Nat.N8.n) at_most
+
+(** [With_length] *)
+module With_length (N : Nat.Intf) : S with type 'a t = ('a, N.n) at_most
+
+(** [of_vector] *)
```
[the remark this answers is not in the document]
Comment (rbonichon) [reply src/lib/pickles_types/at_most.mli]: Done

File: src/lib/pickles_types/nat.mli [hunk 19]
```diff
@@ -0,0 +1,148 @@
...
+(** [Adds] *)
+module Adds : sig
+  type ('a, 'b, 'c) t =
+    | Z : (z, 'n, 'n) t
+    | S : ('a, 'b, 'c) t -> ('a s, 'b, 'c s) t
+
+  val add_zr : 'n nat -> ('n, z, 'n) t
+end
+
+(** [Lte] *)
```
[the remark this answers is not in the document]
Comment (rbonichon) [reply src/lib/pickles_types/nat.mli]: Done

File: src/lib/pickles_types/vector.mli (hunk 12 above)
Comment (rbonichon) [reply src/lib/pickles_types/vector.mli]: Done

File: src/lib/pickles_types/vector.mli (hunk 14 above)
Comment (rbonichon) [reply src/lib/pickles_types/vector.mli]: Done

File: src/lib/pickles_types/vector.mli (hunk 16 above)
Comment (rbonichon) [reply src/lib/pickles_types/vector.mli]: Done

File: src/lib/pickles_types/vector.mli (hunk 15 above)
Comment (rbonichon) [reply src/lib/pickles_types/vector.mli]: Done

**Comment by rbonichon** [comment]
> Looks like the `@` in the branch name is causing an issue with CI. Can you rename the branch? (If you have to open another PR to do so, feel free to ping me and I'll approve it)

Indeed, let me change the '@' to a '/'.

**Pull request closed without merging**